gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.orchestration;
+
+/**
+ * Factory for creating {@link DagProc} based on the visitor type {@link DagTask}.
+ */
+public interface DagProcFactory extends DagTaskVisitor<DagProc> {
+  DagProc meet(LaunchDagTask ldt);
+  DagProc meet(KillDagTask kdt);
+  DagProc meet(ResumeDagTask rdt);
```

Comment:
Let's add the SLA type actions as well from above

-  enforceFlowCompletionDeadline
-   void enforceJobStartDeadline